//  Copyright 2024 Hathor Labs
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.

syntax = "proto3";

package p2p;

service P2PManager {
    rpc IsPeerWhitelisted(BytesValue) returns (BoolValue);
    rpc GetEnabledSyncVersions(Empty) returns (StringList);
    rpc GetVerifiedPeers(Empty) returns (StringList);
    rpc OnReceivePeer(StringValue) returns (Empty);
    rpc OnPeerConnect(StringValue) returns (Empty);
    rpc OnPeerReady(StringValue) returns (Empty);
    rpc OnPeerDisconnect(StringValue) returns (Empty);
    rpc GetRandBytes(Int32Value) returns (BytesValue);
    rpc IsPeerConnected(BytesValue) returns (BoolValue);
    rpc SendTxToPeers(BytesValue) returns (Empty);
}

message Empty {}

message BoolValue {
    bool value = 1;
}

message StringValue {
    string value = 1;
}

message BytesValue {
    bytes value = 1;
}

message Int32Value {
    int32 value = 1;
}

message StringList {
    repeated string values = 1;
}
